### ISSUE TITLE
OpenCodeReasoning-2

### ISSUE THREAD

**Issue by faresobeid** [NVIDIA/NeMo-Skills#559]
Thanks for these great datasets!

Firstly, I noticed the OpenCodeReasoning-2 dataset on huggingface has an empty question column which seems like a mistake. 
Also, was wondering whether currently, OpenCodeReasoning-2 and/or OpenCodeReasoning have r1 or r1-0528 traces as the OpenCodeReasoning-Nemotron-1.1 mentioned it was r1-0528. 
Finally, is there a reason why OpenCodeReasoning-Nemotron-1.1 wasn't trained on OpenCodeReasoning-2?

Thanks once again

**Comment by titu1994** [comment]
> I noticed the OpenCodeReasoning-2 dataset on huggingface has an empty question column which seems like a mistake.

We chose not to redistribute the questions themselves, as the four seed datasets we use themselves gather questions with various licenses. Therefore we provide a short script that users can run, which merges the question from the original datasets into the solutions from our dataset.

> Also, was wondering whether currently, OpenCodeReasoning-2 and/or OpenCodeReasoning have r1 or r1-0528 traces as the OpenCodeReasoning-Nemotron-1.1

Good question. OCR 1 and OCR 2 both are trained on data from the original R1 model. 

We use the major version update to denote significant difference between the two datasets - OCR 1 was a straightforward code completion dataset. OCR 2 is a Code Completion + Code Self Critique dataset, it's goal is to enable research in self critique models. 

OCR 1.1 model is a refresh of the original OCR 1 dataset, just using DeepSeek R1 0528 and we also scaled up the data. 

We plan to release the full OCR 1.1 dataset in the coming month or two. That will enable full replication of the model that is currently released. 

> Finally, is there a reason why OpenCodeReasoning-Nemotron-1.1 wasn't trained on OpenCodeReasoning-2?

Cause 1.1 models are just code generation models, not self critique type models. We could technically add self critique data but we wanted a straightforward upgrade path from 1.0 to 1.1 without any other changes necessary

**Comment by titu1994** [comment]
> Is there a timeline on when the OCR 2 model will be released (or paper)?

I'm hoping in a week or two, we'll  have the paper out. @wasiahmad is working on it. 

Models - I don't think we're planning on releasing them, since we have more recent results from 1.1 OCR that strictly surpasses scores. I suppose there is value in releasing them, but the release process takes a long time and the value gained is not significant from this release. 

> Also was wondering if training on the base vs instruct for both OCR 1.1 and OCR 2 made a difference (compared to OCR 1 where it didn't much for larger models)

We ended up just training instruct models, since it was giving slightly more stable results and ends up being slightly better than base model fine-tuning more often than not. 

> Would OCR 2 without the self critique be a better option to use for SFT than OCR 1.1?

Depends on the goal. If you want just code completion, OCR 1.1 is way better IFF you can afford to infer on 48-64k tokens at inference time. It will give way better scores. However if you want to limit length and still maximize scores, or you want to improve CPP code generation, OCR 2 is the dataset of choice. It's scores will be worse than OCR 1.1, but it will take less tokens and will do better on CPP benchmarks 

> Also could you provide the prompts used for r1 solutions and to obtain QwQ critiques?

Good point. First we are open sourcing how we generated OCR 1 and 2 CPP and python data using old R1. We can eventually release the prompts and script for critique generation. But the paper will have the prompts anyway so if should be easily reproducible, just one LLM call. 

> Additionally, is there a reason to use QwQ for critiques and r1 for answers? 

Time mainly. R1 takes too long to do both. Also our experiments showed QwQ performed on par with R1 for critique step but worse for solution generation step. 

> Would generating both with r1-0528 be the best option?

It might but it's so token hungry. Think about it - you'll need two calls of nearly 64k tokens each, vs two 32k token jobs (and critiques are usually shorter with 20k tokens) 

> Finally, what max CoT length did u use for training compared to distilling from r1

Old R1 - 32k Qwen tokens max 
New R1 - 48k Qwen tokens max

**Comment by faresobeid** [comment]
Thanks for the response!

> Depends on the goal. If you want just code completion, OCR 1.1 is way better IFF you can afford to infer on 48-64k tokens at inference time. It will give way better scores. However if you want to limit length and still maximize scores, or you want to improve CPP code generation, OCR 2 is the dataset of choice. It's scores will be worse than OCR 1.1, but it will take less tokens and will do better on CPP benchmarks

Regarding this, i meant if you collected from r1-0528 for both. Would this still hold?
